src/libtrx/game/items.c

```diff
@@ -60,6 +92,148 @@ int16_t Item_CreateLevelItem(void)
     return item_num;
 }
 
+void Item_Kill(const int16_t item_num)
+{
+    Item_RemoveActive(item_num);
+    Item_RemoveDrawn(item_num);
+
+    ITEM *const item = &m_Items[item_num];
+    LARA_INFO *const lara = Lara_GetLaraInfo();
+    if (item == lara->target) {
+        lara->target = nullptr;
+    }
+
+#if TR_VERSION == 1
+    item->hit_points = -1;
+    item->flags |= IF_KILLED;
+#else
+    if (item_num < m_LevelItemCount) {
+        item->flags |= IF_KILLED;
+    }
```

Comment:
I wonder why we're doing this check in TR2+. I mean, why would the game treat dynamically created objects any different?

Reply:
Yes, it seems a bit odd, but I figured safer for now to keep it as-is. I'll need to study elsewhere to see if the flag is used in different ways possibly first.